src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs

```diff
@@ -399,6 +400,96 @@ public Task GetImport()
             }
         }
 
+        [RavenAction("/databases/*/smuggler/import/csv", "POST", AuthorizationStatus.ValidUser)]
+        public async Task ImportFromCsv()
+        {
+            using (Database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                if (HttpContext.Request.HasFormContentType == false)
+                {
+                    HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+                    using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+                    {
+                        context.Write(writer, new DynamicJsonValue
+                        {
+                            ["Type"] = "Error",
+                            ["Error"] = "Import from csv requires form content type"
+                        });
+                        return;
+                    }
+                }
+                var token = new OperationCancelToken(Database.DatabaseShutdown);
+                var result = new SmugglerResult();
+                var operationId = Database.Operations.GetNextOperationId();
+                await Database.Operations.AddOperation(Database, $"Import to: {Database.Name} from csv file", Raven.Server.Documents.Operations.Operations.OperationType.DatabaseImportFromCsv,
```

Comment:
you await for operation and then result operation id, so this operation id is almost useless. 
Operation id should be taken from request uri - please have a look at import database endpoint. We also supply file and return operation id.

Reply:
I think that the issue is the difference in uploading the file vs. processing the file. But yes, this should be split.

Reply:
I have no idea what you guys mean here, this is exactly how it is done with smuggler import

Reply:
ok you can await it but then operation id should be taken from request url, not generating here. https://github.com/ravendb/ravendb/blob/4b2c605ffa3af88a35ae66b828f5612414a8e3f6/src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs#L331